hub/api/dataset.py

```diff
@@ -343,6 +345,14 @@ def ingest(
             if os.path.samefile(src, dest):
                 raise SamePathException(src)
 
+        if len(os.listdir(src)) < 1:
+            raise AutoCompressionError(src)
+
+        if images_compression == "auto":
+            images_compression = get_most_common_extension(src)
+            if images_compression is None:
+                raise AutoCompressionError(src)
```

Comment:
ideally we should also print the filename for the user and tell them there was no file extension

Reply:
this is scanning an entire directory, so this exception can print the directory that holds the file(s) instead of the exact file. This works on a util file so I'll keep it pretty general something like:

`InvalidFileExtension: Missing file extension in {directory}. Expected a valid file extension got None.`